### PR TITLE
Use a matrix of python versions for the pipeline

### PR DESCRIPTION
**Pull request by thejaminator**
Made the CI use a matrix of python versions so that we don't have to keep remembering to edit the separate pipelines individually

### CODE REVIEW

File: .github/workflows/cpu_ci.yml [hunk 1]
```diff
@@ -3,14 +3,17 @@ name: "Run CPU Tests"
 on: "push"
 
 jobs:
-  run-tests-python3_9:
-    runs-on: ubuntu-latest
+  run-tests:
+    strategy:
+      matrix:
+        python-versions: [ 3.9, "3.10" ]
+        os: [ ubuntu-latest, macos-latest]
```
Comment (thejaminator) [inline .github/workflows/cpu_ci.yml:10]: also  added macos-latest

**Comment by norabelrose** [comment]
Could you resolve the merge conflicts? Will merge after that. Thanks

**Comment by thejaminator** [comment]
Fixed!

File: .github/workflows/cpu_ci.yml (hunk 1 above)
Comment (norabelrose) [reply .github/workflows/cpu_ci.yml:10]: oh that's cool

**Review by norabelrose — APPROVED** [review]
LGTM, I'd like to add Python 3.11 but I think I can do that myself

**Pull request merged**